R/readTechCosts.R

```diff
@@ -59,29 +79,470 @@ readTechCosts <- function(subtype = "PowerAndHeat") { # nolint
+      df[seq(from=index_of_NA[21], to=(nrow(df))) , 9] <- df[index_of_NA[21],1]
+
+      df[,3] <- as.numeric(df[,3])
+      df[,5] <- as.numeric(df[,5])
+      df[,6] <- as.numeric(df[,6])
+      df[,8] <- as.numeric(df[,8])
+
+      df <- df[!is.na(df$'Current'), ]
+      dfp <- pivot_longer(df, cols = c(2:8))
+      #dfp = dfp[ ,-3]
+
+      dfp$period <- NA
+      dfp <- as.data.frame(dfp)
+      dfp[seq(from=1, to=nrow(dfp), by =7) , 5] <- "2020"
+      dfp[seq(from=2, to=nrow(dfp), by =7) , 5] <- "2030"
+      dfp[seq(from=3, to=nrow(dfp), by =7) , 5] <- "2030"
+      dfp[seq(from=4, to=nrow(dfp), by =7) , 5] <- "2030"
+      dfp[seq(from=5, to=nrow(dfp), by =7) , 5] <- "2050"
+      dfp[seq(from=6, to=nrow(dfp), by =7) , 5] <- "2050"
+      dfp[seq(from=7, to=nrow(dfp), by =7) , 5] <- "2050"
+
+      dfp[["name"]] <- sub("Current", "medium", dfp[["name"]])
+      dfp[["name"]] <- sub("2030", "low", dfp[["name"]])
+      dfp[["name"]] <- sub("...4", "medium", dfp[["name"]])
+      dfp[["name"]] <- sub("...5", "high", dfp[["name"]])
+      dfp[["name"]] <- sub("Ultimate", "low", dfp[["name"]])
+      dfp[["name"]] <- sub("...7", "medium", dfp[["name"]])
+      dfp[["name"]] <- sub("...8", "high", dfp[["name"]])
+
+      names(dfp)[3] <- "levels"
+      x <- as.quitte(dfp)
   } else if (subtype == "infrastructure") {
-      x <- read.csv("Infrastructuref2.csv")
-      names(x) <- c("tech", "value", "measurement","variable", "unit", "title")
+
+      df <- read_excel("REF2020_Technology Assumptions_Transport.xlsx", sheet = "Infrastructure", range = "B3:H17")
+
+      df <- as.data.frame(df)
+      df2 <- df
+      df = df[-1,] # remove first row
+
+      df[,4] <- as.numeric(df[,4])
+      df[,7] <- as.numeric(df[,7])
+
+      dfp <- pivot_longer(df, cols = c(2:7))
+
+      dfp$period= NA
+      dfp <- as.data.frame(dfp)
+      dfp[seq(from=1, to=nrow(dfp), by=3) , 4] <- df2[1,2]
+      dfp[seq(from=2, to=nrow(dfp), by=3) , 4] <- df2[1,3]
+      dfp[seq(from=3, to=nrow(dfp), by=3) , 4] <- df2[1,4]
+
+      dfp$variable= NA
+      dfp <- as.data.frame(dfp)
+      dfp[seq(from=1, to=nrow(dfp), by=6) , 5] <- names(df[2])
+      dfp[seq(from=2, to=nrow(dfp), by=6) , 5] <- names(df[2])
+      dfp[seq(from=3, to=nrow(dfp), by=6) , 5] <- names(df[2])
+      dfp[seq(from=4, to=nrow(dfp), by=6) , 5] <- names(df[5])
+      dfp[seq(from=5, to=nrow(dfp), by=6) , 5] <- names(df[5])
+      dfp[seq(from=6, to=nrow(dfp), by=6) , 5] <- names(df[5])
+
+
+      dfp[["period"]] <- sub("Ultimate", "2050", dfp[["period"]])
+      dfp <- dfp[,-2]
+
+      x <- as.quitte(dfp)
       x[["value"]] <- as.numeric(x[["value"]])
       x <- as.quitte(x)
+
   } else if (subtype == "new_fuels_energy") {
-      x <- read.csv("new_fuels_energy.csv")
-      x <- as.quitte(x)
+    df <- read_excel("REF2020_Technology Assumptions_Energy.xlsx", sheet = "New Fuels", range = "A3:G23")
+
+    df <- as.data.frame(df)

```

Comment:
why are you converting to data.frame here? isn't `df` already a data.frame?

Reply:
it is "list", how do you want me to proceed?

Reply:
it's fine as it is if `df` is a list